src/parser/deserializer.ts

```diff
@@ -433,7 +433,8 @@ function deserializeObject(
         if (promoteBuffers && promoteValues) {
           value = _buffer;
         } else if (subType === constants.BSON_BINARY_SUBTYPE_UUID_NEW) {
-          value = new Binary(buffer.slice(index, index + binarySize), subType).toUUID();
+          const binary = new Binary(buffer.slice(index, index + binarySize), subType);
+          value = UUID.isValid(binary) ? binary.toUUID() : binary;
         } else {
           value = new Binary(buffer.slice(index, index + binarySize), subType);
         }
```

Comment:
```suggestion
          value = new Binary(buffer.slice(index, index + binarySize), subType);
          if (subType === constants.BSON_BINARY_SUBTYPE_UUID_NEW && UUID.isValid(value)) {
                      value = value.toUUID();
          }
```
can we align the logic between deserialize and deserializeObject?

Reply:
good cleanup, fixed